test/e2e/apiextensions_test.go

```diff
@@ -184,3 +184,35 @@ func TestCompositionFunctions(t *testing.T) {
 			Feature(),
 	)
 }
+
+// TestXRDDefaults checks if claims fields are defaulted by k8s API
+// according to defined XRD spec
+func TestXRDDefaults(t *testing.T) {
```

Comment:
I think it is, otherwise, how we could be sure that apiserver sets the values, and not the controller.

Reply:
I think we can trust that if we generate the CRD properly as you are checking in the CRD the API server will behave properly

Reply:
Trust is good, but verifying is always better :)

The unit tests does not checking anything about the validity of the given CRD. While working on this PR, I had the test passing, but after deploying it to the cluster, I was not able to create claims, because I did not craft schema default field properly. Only e2e tests can catch such things.

So far, the claim controller was setting defaults, but given that we here set the defaults on CRD, that logic was removed from the controller. 

As an user, I would expect that the created claim gets initialized all defaults even before the first reconciliation. Thus, the added e2e tests creates a claim in paused state.

Prior to this PR, the added tests would fail. In the future, it would be able to catch regressions.

Reply:
one can see this tests as the beginning of whole suite around https://github.com/crossplane/crossplane/issues/4619

Reply:
e2es are to be considered expensive, first because they take time and second because given that we share a single environment atm, they can be brittle, we should fix that, but in the meantime we should take that into account. Here you are just checking that the generated CRD is valid, which you could do even without a provider and a composition, just deploying the XRD and a claim. However I feel it would be a better fit to use something like [envtest](https://book.kubebuilder.io/reference/envtest.html) for such a check, without having to spin up a whole cluster to do that.

Reply:
> which you could do even without a provider and a composition, just deploying the XRD and a claim. 

Sure, we can make the test setup lighter. Should I do it?

> However I feel it would be a better fit to use something like [envtest](https://book.kubebuilder.io/reference/envtest.html) for such a check, without having to spin up a whole cluster to do that.

`envtest` is an integration suite, and iirc we do not want integration tests in this project, either unit or e2e tests. See https://github.com/crossplane/crossplane/pull/4261#discussion_r1240486106

Reply:
> envtest is an integration suite, and iirc we do not want integration tests in this project, either unit or e2e tests. See #4261 (comment)

We just don't want to introduce a new category of tests lightly and without discussing it first 😉

Yes, at least make it minimal if you really feel having an e2e for it makes sense.

Reply:
> We just don't want to introduce a new category of tests lightly and without discussing it first 😉

Sure, and therefore I cannot introduce it in this PR.

> Yes, at least make it minimal if you really feel having an e2e for it makes sense.

Given that I would like to unblock merging of this PR quickly, and with the strong review feedback that the added e2e tests is unnecessary, I have removed it.